src/error.rs

```diff
@@ -148,6 +148,7 @@ pub enum DbOptionsError {
     #[error("Unknown configuration file format: {0}")]
     UnknownFormat(PathBuf),
 
+    // TODO: clippy complains this error is too large
```

Comment:
this clippy warning on error struct size may propagate to users of the `slatedb` crate, all users with the newer toolchain will get clippy error by simply referencing `SlateDBError` unless setting `#![allow(clippy::result_large_err)]`.

i think we can fix this size error in another pr. 🤔